mycroft/client/enclosure/generic/__init__.py

```diff
@@ -41,7 +41,8 @@ def __init__(self):
         super().__init__()
 
         # Notifications from mycroft-core
-        self.bus.on("enclosure.notify.no_internet", self.on_no_internet)
+        self.bus.on('enclosure.notify.no_internet', self.on_no_internet)
+        self.bus.on('mycroft.skills.trained', self.is_device_ready)
```

Comment:
Hmm, this is something to think about. Either that or start polling readiness directly without any previous message making the skill process require training before setting ready...

The reason the polling system was setup (instead of simple readiness status emits) was to allow any startup order, but this sort of requires the enclosure to be running before the skills are loaded.

Reply:
I think this was made to mirror the current behaviour where the mycroft.ready message depended directly on padatious training so I can see this as a good first implementation. since it mirrors the current workflow and in basically all cases we directly care about it works as expected.

Reply:
Please add a TODO that indicates this is a potential pitfall if something runs longer than Padatious training.